src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,65 @@
+package step2;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final Pattern STRING_PATTERN = Pattern.compile("//(.)\n(.*)");
+
+    public static int splitAndSum(String data) {
+        if (data == null || data.isBlank()) {
+            return 0;
+        }
+
+        String pattern = findPattern(data);
+        if (!pattern.isBlank()) {
+            String[] strings = getNumbers(data, pattern);
+            int result = 0;
+            for (String str : strings) {
+                int num = Integer.parseInt(str);
+                checkPositive(num);
+
+                result += num;
+            }
+
+            return result;
+        }
+
+        int aNumber = Integer.parseInt(data);
+        checkPositive(aNumber);
+
+        return aNumber;
+    }
+
+    private static String findPattern(String data) {
+        String delimiter = "";
+        Matcher m = STRING_PATTERN.matcher(data);
+        if (m.find()) {
+            return m.group(1);
+        }
+
+        try {
+            delimiter = data.split(",|:").length > 0 ? ",|:" : "";
+        } catch (Exception e) {
+            // nothing to do
+        }
+
+        return delimiter;
+    }
+
+    private static String[] getNumbers(String data, String pattern) {
+        Matcher m = STRING_PATTERN.matcher(data);
+        if (m.find()) {
+            return m.group(2).split(pattern);
+        }
+
+        return data.split(",|:");
+    }
+
+    private static void checkPositive(int aNumber) {
+        if (aNumber < 0) {
+            throw new RuntimeException();
+        }
+    }
```

Comment:
```suggestion
    private static void checkPositive(int number) {
        if (number < 0) {
            throw new IllegalArgumentException("음수는 허용하지 않습니다. - " + number);
        }
    }
```
aNumber가 뭘까요? 🤔 
의미가 없다면 number로 사용해도 될 것 같습니다.

또한 `RuntimeException` 대신 적절한 표준 예외를 사용하면 어떨까요?
매개변수 값이 잘못되었다는 예외인 `IllegalArgumentException`에 메시지까지 담아 예외를 던지면 좋을 것 같아요
<img width="522" alt="image" src="https://github.com/next-step/java-racingcar/assets/36617783/54b8d070-476a-4edd-bf41-94cbe5db6754">